python/ray/tune/trainable/trainable.py

```diff
@@ -546,13 +547,16 @@ def save(
                 metrics = self._last_result.copy() if self._last_result else {}
 
                 if self._storage:
+                    # The checkpoint index is updated with the current result.
+                    # NOTE: This is no longer using "iteration" as the folder indexing
+                    # to be consistent with fn trainables.
+                    self._storage._update_checkpoint_index(metrics)
+
                     persisted_checkpoint = self._storage.persist_current_checkpoint(
                         local_checkpoint
                     )
-                    # The checkpoint index needs to be incremented.
-                    # NOTE: This is no longer using "iteration" as the folder indexing
-                    # to be consistent with fn trainables.
-                    self._storage._increase_checkpoint_index(metrics)
+
+                    metrics[CHECKPOINT_DIR_NAME] = self._storage.checkpoint_dir_name
```

Comment:
Updating the metrics here will not get saved to the `result.json` -- this is generally ok, except for `Result.from_path` which relies on this metric to construct the list of checkpoints _with their metrics attached_

The tests don't currently cover `Result.from_path` usage with cls trainable.

Reply:
Ah you're right, I've removed it for now